### PR TITLE
Destroy empty workspaces when moving away

### PR DESCRIPTION
**Pull request by SirCmpwn**

### CODE REVIEW

File: sway/tree/container.c [hunk 1]
```diff
@@ -236,6 +235,45 @@ struct sway_container *container_output_destroy(struct sway_container *output) {
 	return &root_container;
 }
 
+struct sway_container *container_workspace_destroy(
```
Comment (acrisci) [inline sway/tree/container.c]: probably belongs in workspace.c

Comment (SirCmpwn) [reply sway/tree/container.c]: We already have `container_output_destroy` and `container_view_destroy` in this file, I'll move them all

File: sway/tree/layout.c [hunk 2]
```diff
@@ -121,6 +122,42 @@ struct sway_container *container_remove_child(struct sway_container *child) {
 	return parent;
 }
 
+struct sway_container *container_reap_empty(struct sway_container *container) {
```
Comment (acrisci) [inline sway/tree/layout.c]: container_coalesce?

File: sway/input/seat.c [hunk 3]
```diff
@@ -331,6 +332,9 @@ void sway_seat_set_focus(struct sway_seat *seat, struct sway_container *containe
 		if (last_ws) {
 			wlr_log(L_DEBUG, "sending workspace event");
 			ipc_event_workspace(last_ws, container, "focus");
+			if (last_ws->children->length == 0) {
```
Comment (acrisci) [inline sway/input/seat.c]: let's have `container_num_children()` otherwise we need a todo for floating.

**Comment by acrisci** [comment]
:+1: looks ok so far

**Comment by acrisci** [comment]
how to test?

**Comment by emersion** [comment]
>how to test?

Open views in workspaces, close all views on one workspace, switch to another one. The empty workspace should be deleted.

**Comment by SirCmpwn** [comment]
Shit wrong PR to merge

**Comment by SirCmpwn** [comment]
I'll clean it up in another PR

**Pull request merged**